src/include/random.h

```diff
@@ -141,6 +153,11 @@ int generate_random_number(MutexT& m, EngineT& e)
 
 } // namespace detail
 
+bool rng_is_seeded() 
+{
+  return detail::rng_is_seeded_state();
+}
```

Comment:
That's correct, though hopefully at known points in the application. The issue I have with putting the test inside the generator is that you're checking a flag for each number you want, every time. This isn't going to be great in a tight loop where you want a lot of random numbers.